rasa_sdk/forms.py

```diff
@@ -242,8 +264,8 @@ def extract_other_slots(
                     # by entity with the same name
                     should_fill_entity_slot = (
                         other_slot_mapping["type"] == "from_entity"
-                        and other_slot_mapping.get("entity") == slot
                         and self.intent_is_desired(other_slot_mapping, tracker)
+                        and self.slot_is_desired(other_slot_mapping, slot, tracker)
```

Comment:
This is a bit confusing to me -- the slot is always desired if it's empty. Should this be `entity_is_desired`? I.e. I guess what you are changing here is that even if the entity_type from the other mapping matches name of the slot, that does not necessarily mean that the entity is desired unless the group/role is right as well. Am i interpreting that correctly?

Reply:
Yeah, `entity_is_desired` is a better name. So either the slot name is equal to the entity name (normal auto filling) or the slot mapping triggers (to allow to auto-fill slots with entities that have a role/group).